### ISSUE TITLE
logarithmic axis with max shows wrong data

### ISSUE THREAD

**Issue by tmtron** [apache/echarts#15329]
### Version
5.1.2

### Reproduction link
[https://stackblitz.com/edit/echarts5-log-max-issue?file=opt1.js](https://stackblitz.com/edit/echarts5-log-max-issue?file=opt1.js)

### Steps to reproduce
* use a yAxis of type='log' and set the max value lower than the lowest value in the data

### What is expected?
the y-Axis should show up and no data should be displayed
![image](https://user-images.githubusercontent.com/20422904/125264888-9559d980-e304-11eb-83d3-404446f03010.png)

### What is actually happening?
the y-Axis is not shown and all data is displayed:  
![image](https://user-images.githubusercontent.com/20422904/125264690-5deb2d00-e304-11eb-9f83-70541c566337.png)

Notes:
* the lowest value in the data is 100k (100_000)
* the y-Axis mas is set to 10k (10_000)
* so the chart should not display any data
* the y-Axis should still be displayed, since we have set a maximum

### Additional info
Here is the full data (when we don't set min/max):  
![image](https://user-images.githubusercontent.com/20422904/125264990-ae628a80-e304-11eb-8021-3e18d6fcf3fe.png)

**Comment by tmtron** [comment]
I just noticed, that this also happens with y-Axis type = 'value'.  
[Stackblitz example with linear axis](https://stackblitz.com/edit/echarts5-linear-max-issue?file=opt1.js)

**Comment by susiwen8** [comment]
Yes, if `min` were not set, echarts would take the min of data as min of data
https://github.com/apache/echarts/blob/master/src/coord/scaleRawExtentInfo.ts#L190-L208

so if `max` were lower than data's min then final `min` and `max` would turn into `NaN`

**Comment by tmtron** [comment]
in Version 5.2.2 the example rendering looks like this:

![image](https://user-images.githubusercontent.com/20422904/139644224-9070a179-7d25-4ec8-916b-90c995f5605b.png)


see [Stackblitz example V5.2.2](https://stackblitz.com/edit/echarts5-log-max-issue-uwogk1?file=opt1.js)

**Comment by tmtron** [comment]
same issue in V5.4.3 - [Stackblitz example](https://stackblitz.com/edit/echarts5-log-max-issue-f5vkrq?file=opt1.js)
